src/api/entities/HistoricalMultiSigProposal.ts

```diff
@@ -0,0 +1,177 @@
+import BigNumber from 'bignumber.js';
+
+import { Context, Entity } from '~/internal';
+import { multiSigProposalVotesQuery } from '~/middleware/queries';
+import { Query } from '~/middleware/types';
+import { MultiSigProposalVote, SignerType } from '~/types';
+import { Ensured } from '~/types/utils';
+import { middlewareEventDetailsToEventIdentifier, signerValueToSigner } from '~/utils/conversion';
+
+export type MiddlewareMultiSigProposal = {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
+  creator?: {
+    did: string;
+  };
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: Date;
+};
+
+type MiddlewareMultiSigProposalUniqueIdentifiers = Pick<MiddlewareMultiSigProposal, 'id'>;
+
+export interface HumanReadable {
+  id: string;
+  proposalId: number;
+  multisigId: string;
+  approvalCount: number;
+  rejectionCount: number;
+  creatorDid?: string;
+  status: string;
+  createdBlockId: string;
+  updatedBlockId: string;
+  datetime: string;
+}
+
+enum MultiSigStatusEnum {
+  Pending = 'Pending',
+  Success = 'Success',
+  Failed = 'Failed',
+}
+
+/**
+ * Represents historical MultiSigProposal that no longer exists on chain
```

Comment:
```suggestion
 * Represents a historical MultiSigProposal
```

(nit) I think these are made in SQ when the propsal is added, so it could exist on chain still. Maybe mention this will persist after execution instead.